### PR TITLE
[move-lang][move-model] Gather used addresses, including in values

### PR DESCRIPTION
**Pull request by tnowacki**
- Taking a look at named addresses again, support for values used in addresses is missing
- Kept track of the addresses used in dependency_ordering

## Motivation

- Looking again at landing #8394 and this was missing in modules that used them only in specs or as a value 

## Test Plan

- Will come up in #8394

### CODE REVIEW

File: language/move-lang/src/expansion/ast.rs [hunk 1]
```diff
@@ -453,6 +455,7 @@ impl Address {
             Self::Named(n) => match addresses.get(&n) {
                 Some(a) => Ok(*a),
                 None => {
+                    panic!();
```
Comment (mengxu-fb) [inline language/move-lang/src/expansion/ast.rs:458]: oops?

**Review by mengxu-fb — APPROVED** [review]

**Comment by tnowacki** [comment]
This hasn't quite fixed everything yet... and I'm not sure why. Debugging now

**Comment by tnowacki** [comment]
- Forgot to actually update the used_address set... whoops
- Did not grab transitive closure

**Review by mengxu-fb — DISMISSED** [review]

**Pull request merged**